src/pages/home/uploadSong/UploadSong.tsx

```diff
@@ -84,6 +83,16 @@ const UploadSong: FunctionComponent = () => {
 
   // eslint-disable-next-line
   const handleSubmit = (values: UploadSongRequest) => {
+    generateArtistAgreement({
+      body: {
+        songName: values.title,
+        companyName,
+        artistName,
+        stageName,
+        songId: id,
+        save: true,
+      },
+    });
```

Comment:
So even if the person doesn't select minting (conditional in `handleSongInfo`) the agreement gets generated? or is this part of the initial backend testing?

Reply:
Good catch, forgot to wrap it in a condition :)